source/opt/workaround_image_operands.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "workaround_image_operands.h"
+
+#include <list>
+#include <stack>
+
+namespace spvtools {
+namespace opt {
+
+Pass::Status DeanonymizeImages::Process(ir::IRContext* c) {
+  InitializeProcessing(c);
+  bool modified = false;
+  modified = FixupOpTypeImage();
+  return (modified ? Status::SuccessWithChange : Status::SuccessWithoutChange);
+}
+
+bool DeanonymizeImages::FixupOpTypeImage() {
+  auto* def_use_mgr = context()->get_def_use_mgr();
+  auto* type_mgr = context()->get_type_mgr();
+
+  bool modified = false;
+  auto* module = get_module();
+
+  std::vector<analysis::Image*> unknown_depth_images;
+
+  for (auto type : *type_mgr) {
```

Comment:
Should I even be using the type_mgr here? Or should I just deal directly with  ir::Instruction*s.
The typeMgr might be overkill?